lib/todo.rb

```diff
@@ -29,6 +29,11 @@ def display
       display
     end
 
+    def item_already_added?(item)
+      current_data = get_contents
+      return true if current_data.index(item) != nil
```

Comment:
This is technically correct but idiomatically not Ruby.  This would be written using the `nil?` method instead of a nil comparison, in one of the following ways:

``` ruby
return true if !current_data.index(item).nil?
return true unless current_data.index(item).nil?
``` 

However, we can simplify this line even further.  When you return a `nil` to a conditional like `if`, the `nil` is treated as `false`, and anything else be treated as `true`.  Thus, `current_data.index(item)` will return a value when there's a dupe, and nil when there isn't.

If you don't like that it is not explicitly returning a boolean, we can use a trick to turn that into a boolean.  Prefixing an object with `!` causes Ruby to evaluate it as a conditional statement and return a boolean, but in the opposite: if the var has a value, it will return `false` and if it's `nil` or `false` it will return `true`.  However...if we apply _another_ `!` we can invert _that_ boolean and result in a true boolean response.

That's a lot of words, however, so let me show you an IRB example to help solidify the concept:

```
>> x = "foo"
=> "foo"
>> y = nil
=> nil
>> !x
=> false
>> !y
=> true
>> !!x
=> true
>> !!y
=> false
```

Applied concretely to this, we can express this as `!!current_data.index(item)` and we end up with a Boolean based upon whether we get a value back from `index` or not.

Reply:
Understood